src/policy_artifacthub.rs

```diff
@@ -0,0 +1,923 @@
+    readme: Option<String>,
+}
+
+#[derive(Deserialize, Serialize, Debug, Clone)]
+#[serde(rename_all = "camelCase")]
+struct ContainerImage {
+    /// name is always "policy"
+    name: ConstContainerImageName,
+    /// URL of the policy wasm module
+    /// E.g: ghcr.io/kubewarden/policies/verify-image-signatures:v0.2.1
+    image: String,
+}
+
+#[derive(Deserialize, Serialize, Debug, Clone)]
+enum ConstContainerImageName {
+    #[serde(rename = "policy")]
+    Policy,
+}
+
+/// Link, of either the policy wasm module or the policy source repository
+/// Example:
+///  - name: policy
+///    url: https://github.com/kubewarden/verify-image-signatures/releases/download/v0.2.1/policy.wasm
+///  - name: source
+///    url: https://github.com/kubewarden/verify-image-signatures
+#[derive(Deserialize, Serialize, Debug, Clone, PartialEq)]
+struct Link {
+    /// Either "policy" or "source"
+    name: ConstLinkName,
+    /// Either URL of policy repository or URL of policy wasm module
+    url: Url,
+}
+
+#[derive(Deserialize, Serialize, Debug, Clone, PartialEq)]
+enum ConstLinkName {
+    #[serde(rename = "policy")]
+    Policy,
+    #[serde(rename = "source")]
+    Source,
+}
+
+/// Hardcoded recommendation with url of kubewarden controller
+#[derive(Deserialize, Serialize, Debug, Clone)]
+#[serde(rename_all = "camelCase")]
+struct Recommendation {
+    url: ConstRecommendation,
+}
+
+#[derive(Deserialize, Serialize, Debug, Clone)]
+enum ConstRecommendation {
+    #[serde(rename = "https://artifacthub.io/packages/helm/kubewarden/kubewarden-controller")]
+    Kubewarden,
+}
+
+#[derive(Deserialize, Serialize, Debug, Clone, PartialEq)]
+#[serde(rename_all = "camelCase")]
+struct Maintainer {
+    name: String,
+    email: String,
+}
+
+/// Hardcoded provider with "kubewarden"
+#[derive(Deserialize, Serialize, Debug, Clone)]
+#[serde(rename_all = "camelCase")]
+struct Provider {
+    name: ConstProvider,
+}
+
+#[derive(Deserialize, Serialize, Debug, Clone)]
+enum ConstProvider {
+    #[serde(rename = "kubewarden")]
+    Kubewarden,
+}
+
+impl ArtifactHubPkg {
+    pub fn from_metadata(
+        metadata: &Metadata,
+        version: &str,
+        created_at: OffsetDateTime,
+        questions: Option<&str>,
+    ) -> Result<Self> {
+        // validate inputs
+        //
+        if metadata.annotations.is_none() {
+            return Err(anyhow!(
+                "no annotations in policy metadata. policy metadata must specify annotations"
+            ));
+        }
+        let metadata_annots = metadata.annotations.as_ref().unwrap();
+        if metadata_annots.is_empty() {
+            return Err(anyhow!(
+                "no annotations in policy metadata. policy metadata must specify annotations"
+            ));
+        }
+        let semver_version = Version::parse(version)
+            .map_err(|e| anyhow!("policy version must be in semver: {}", e))?;
+        if questions.is_some() && questions.unwrap().is_empty() {
```

Comment:
Done in https://github.com/kubewarden/policy-evaluator/pull/246/commits/cf93ea3c65ce796f394c89001b6b522a77ae7e11.
I'm ok leaving passing questions as a `Some(s)` because it's optional, so then `parse_annotations()` can see if they are there, and add the corresponding annotation for them or not.